lib/src/flutter_downloader.dart

```diff
@@ -0,0 +1,117 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_downloader/flutter_downloader.dart';
+import 'package:flutter_downloader/src/desktop_platform_download.dart';
+import 'package:flutter_downloader/src/platform_download.dart';
+
+part 'legacy_api.dart';
+
+/// The target where the download should be visible for the user.
+enum DownloadTarget {
```

Comment:
It'd be great if you added what methods can be used to access files under each of these paths.

For example, explain that [path_provider.getDownloadsDirectory()](https://pub.dev/documentation/path_provider/latest/path_provider/getDownloadsDirectory.html) can be used to access the file if its download target is `downloadsFolder`.